### PR TITLE
[WIP] Creating configuration for the rmc

### PR DESCRIPTION
**Pull request by upatricck**
Modified the add new site to support the option to choose between the rmc and the vrm,
also now we are creating the rmc account after creating the rmc account,

Also now adding sensors to the site.

## To be done
* Automatically generatinng the configuration file for the site.

### CODE REVIEW

**Comment by jlark** [comment]
thanks @upatricck can you add some screenshots of the forms?

**Comment by upatricck** [comment]
Some Screen shots

screenshot from 2016-07-14 11 48 00
screenshot from 2016-07-14 11 47 19

**Comment by jlark** [comment]
LGTM. Should we merge this or is it a WIP?

**Comment by upatricck** [comment]
Still a work in progress, I think we should merge it when the config is being generated

**Comment by jlark** [comment]
ok, I agree changed name  accordingly.

File: seshdash/tests/test_rmc_site_creation.py [hunk 1]
```diff
@@ -92,3 +99,17 @@ def test_rmc_account(self):
         self.assertEqual(Sensor_EmonTx.objects.last().site, self.site)
 
 
+    def test_get_rmc_config(self):
+        """
+        Testing the generation of the rmc config file for a 
+        given file
+        """
+
+        response = self.client.post('/get_rmc_config', {'api_key': 'testing'})
```
Comment (jlark) [inline seshdash/tests/test_rmc_site_creation.py]: can you make this a GET request instead of a POST?

File: seshdash/utils/model_tools.py [hunk 2]
```diff
@@ -169,3 +169,17 @@ def associate_sensors_to_site(sensors_list, site):
...
 
 
+def get_associated_sensors(site):
+    """
+    Returns the associated sensors for
+    a given site
+    """
+    sensors_list = []
+    
+    for model in [Sensor_EmonTx, Sensor_EmonTh, Sensor_BMV]:
```
Comment (jlark) [inline seshdash/utils/model_tools.py]: Maybe you can create a SENSORS global variable at the top of the class

File: seshdash/tests/test_rmc_site_creation.py (hunk 1 above)
Comment (upatricck) [inline seshdash/tests/test_rmc_site_creation.py]: Yes, going to.

File: seshdash/tests/test_rmc_site_creation.py [hunk 3]
```diff
@@ -105,9 +107,7 @@ def test_get_rmc_config(self):
...
         """
 
-        response = self.client.post('/get_rmc_config', {'api_key': 'testing'})
-
-
-        print "The status code is: ",
-        print response.status_code
+        response = self.client.get('/get_rmc_config', {'api_key': 'testing'})
 
+        self.assertEqual(response.status_code, 200)
```
Comment (jlark) [inline seshdash/tests/test_rmc_site_creation.py]: Can you also test if a specific node you added to the site is there?

File: seshdash/utils/model_tools.py [hunk 4]
```diff
@@ -183,3 +185,68 @@ def get_associated_sensors(site):
...
+    """
+    Function to generate the configuration for all of the
+    sensors
+    """
+    # Initializing 
+    configuration = ""
+    emonth_range = [5, 6, 7, 8]
+    emonth_index = 0
+    emontx_range = [20, 21, 22]
+    emontx_index = 0
```
Comment (jlark) [inline seshdash/utils/model_tools.py]: what is the index for?

File: seshdash/utils/model_tools.py [hunk 5]
```diff
@@ -183,3 +185,68 @@ def get_associated_sensors(site):
...
+        
+        if type(sensor) is Sensor_EmonTh: 
+            try:
+                number = emonth_range[emonth_index]
+                t = get_template('seshdash/configs/emon_th.conf')
+                text = t.render({'number': number, 'sensor_number': (emonth_index + 1) })
+                emonth_index = emonth_index + 1
+            except IndexError:
+                print "Emon ths sensors out of range, Improperly configured"
+                logger.debug("Emon ths sensors out of range, Improperly configured")
```
Comment (jlark) [inline seshdash/utils/model_tools.py]: should use logger error instead.

File: seshdash/models.py [hunk 6]
```diff
@@ -550,6 +551,89 @@ class Meta:
...
+     """
+     NODE_ID_CHOICES = (
+                    (5, 5),
+                    (6, 6),
+                    (7, 7),
+                    (8, 8),
+               )
+
+     site = models.ForeignKey(Sesh_Site)
+     node_id = models.IntegerField(default=0, choices=NODE_ID_CHOICES)
```
Comment (jlark) [inline seshdash/models.py]: swich the default values with the BMV

File: seshdash/tests/test_rmc_site_creation.py [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+    def test_get_rmc_config(self):
+        """
+        Testing the generation of the rmc config file for a
+        given file
+        """
+
+        response = self.client.get('/get_rmc_config', {'api_key': 'testing'})
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.get('Content-Type'), 'text/plain')
```
Comment (jlark) [inline seshdash/tests/test_rmc_site_creation.py]: `assertRegexpMatches(config_file, 'emonTx_1')`

File: seshdash/views.py [hunk 8]
```diff
@@ -898,33 +902,123 @@ def edit_site(request,site_Id=1):
...
+
+@login_required
+@permission_required_or_403('auth.view_Sesh_Site')
+def add_rmc_account(request, site_id):
+    """
+    This view is for adding a new rmc account
+    to the database and associate it with a site
+    """
+    site = Sesh_Site.objects.filter(id=site_id).first()
+
```
Comment (jlark) [inline seshdash/views.py]: `sensor_arr = ['EmonTh', 'EmonTx', 'BMV']
fomset_dict = {}
for sensor in sensor_arr:
     formset_dict[sensor] = formset_factory('Sensor'+sensor+'Form')

     
`

**Comment by jlark** [comment]
perhaps just add the test for checking config, other than that look s great.

**Pull request merged**